EmailValidator/Validation/Error/IllegalMailbox.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Egulias\EmailValidator\Validation\Error;
+
+use Egulias\EmailValidator\Exception\InvalidEmail;
+
+class IllegalMailbox extends InvalidEmail
+{
+    const CODE = 995;
+    const REASON = "The mailbox is illegal.";
+
+    /**
+     * @var int
+     */
+    private $responseCode;
+
+    /**
+     * IllegalMailbox constructor.
+     *
+     * @param int $responseCode
+     */
+    public function __construct($responseCode)
+    {
+        parent::__construct();
+
+        $this->responseCode = $responseCode;
```

Comment:
In this place private var is necessary because `responseCode` contains SMTP response code https://www.greenend.org.uk/rjk/tech/smtpreplies.html

Users can understand the reason, why mailbox is illegal.

Reply:
You are right.
What happens is that `InvalidEmail` does not provide a method for accessing `responseCode`, which is private. Thus, user won't be able to get the code.
The only way for user, AFAIK, to get the code would be to overwrite intherited `__toString` (see https://www.php.net/manual/en/class.exception.php and https://www.php.net/manual/en/language.exceptions.extending.php), the only method allowed to be overwritten:
```php
public function __toString()
{
  return ": SMTP response code was {$this->responseCode}, exception reason,code {$this->message} - {$this->code}";
}
```
Another solution would be create an empty interface `EmailValidationException`, make `InvalidEmail` implement it and then 
```php
InvalidMailbox extends \InvalidArgumentException implements EmailValidationException
{
    private $reasons = [550 => 'Requested action not taken: mailbox unavailable', 551 => ...]
    public function __construct($responseCode)
    {
        $selectedReason = isset($reasons[$responseCode]) ? $reasons[$responseCode] : "Unknown reason";
        parent::__construct($selectedReason, $responseCode);
    }
}
```
And adapt all the necesary code to allow for this new type of exception by requiring the interface `EmailValidationException` instead of the concrete class `InvalidEmail`.
```
What do you think?